File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        Set<String> set = new HashSet<>();
        set.add(words[0]); 
        char end = words[0].charAt(words[0].length() - 1);
        for (int i = 1; i < words.length; i++) {
            if(end != words[i].charAt(0) || set.contains(words[i])){
                answer[0] = i % n + 1;
                answer[1] = i / n + 1  ;
                break;
            }
            set.add(words[i]);
            end = words[i].charAt(words[i].length() - 1);
        }

        return answer;
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    var answer = [];
    for(let i = 0 ; i<words.length ; i++){
        let num = i%n ; 
        let strs = answer[answer.length-1] ?? words[0][0] ;  
       
        
        if(words[i].length === 1 || answer.indexOf(words[i]) !== -1 || strs.charAt(strs.length-1) !== words[i][0]){
           
            return [num+1 , Math.floor(answer.length/n)+1] ; 
        }
       
        
        answer.push(words[i]) ;
    }
 

    return [0,0];
}
    
